app/src/components/core/model/toolboxElementControl.ts

```diff
@@ -16,13 +17,22 @@ module app.core.model {
             }
             this.elementConfig = config;
             this.type = type;
+        }
 
+        generateLabel(name) {
```

Comment:
Rather return string here instead of assigning the value directly to the label. Keep assignment logic in the constructor.